app/uk/gov/hmrc/cardpaymentfrontend/models/extendedorigins/ExtendedNiEuVatIoss.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cardpaymentfrontend.models.extendedorigins
+
+import payapi.cardpaymentjourney.model.journey.{JourneySpecificData, JsdNiEuVatIoss}
+import play.api.i18n.Messages
+import play.api.mvc.AnyContent
+import uk.gov.hmrc.cardpaymentfrontend.actions.JourneyRequest
+import uk.gov.hmrc.cardpaymentfrontend.models.PaymentMethod._
+import uk.gov.hmrc.cardpaymentfrontend.models.openbanking.{NiEuVatIossSessionData, OriginSpecificSessionData}
+import uk.gov.hmrc.cardpaymentfrontend.models.{CheckYourAnswersRow, PaymentMethod}
+import uk.gov.hmrc.cardpaymentfrontend.util.Period.displayCalendarPeriodMonth
+
+object ExtendedNiEuVatIoss extends ExtendedOrigin {
+  override val serviceNameMessageKey: String = "service-name.NiEuVatIoss"
+  override val taxNameMessageKey: String = "payment-complete.tax-name.NiEuVatIoss"
+
+  def cardFeesPagePaymentMethods: Set[PaymentMethod] = Set(OpenBanking)
+  def paymentMethods(): Set[PaymentMethod] = Set(Card, OpenBanking, Bacs)
+
+  override def checkYourAnswersReferenceRow(journeyRequest: JourneyRequest[AnyContent])(payFrontendBaseUrl: String): Option[CheckYourAnswersRow] = {
+    journeyRequest.journey.journeySpecificData.reference.map { vrn =>
+      CheckYourAnswersRow(
+        titleMessageKey = "check-your-details.NiEuVatIoss.reference",
+        value           = Seq(vrn.value),
+        changeLink      = None
+      )
+    }
+  }
+
+  override def checkYourAnswersAdditionalReferenceRow(journeyRequest: JourneyRequest[AnyContent])(payFrontendBaseUrl: String)(implicit messages: Messages): Option[Seq[CheckYourAnswersRow]] = {
+    val period = journeyRequest.journey.journeySpecificData.asInstanceOf[JsdNiEuVatIoss].period
```

Comment:
Can you try and avoid using `asInstanceOf` in prod code, it can throw `ClassCastException` which would be uncaught and unexpected. Instead we should use pattern match on Jsd and throw our own expected exception - not much in it technically, but asInstanceOf is code smell IMO